cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/polling/ClouderaManagerPollingServiceProvider.java

```diff
@@ -157,6 +167,19 @@ public PollingResult startPollingCmApplyHostTemplate(Stack stack, ApiClient apiC
         return startDefaultPolling(stack, apiClient, commandId, "Apply host template");
     }
 
+    public PollingResult startPollingCmHostsRecommission(Stack stack, ApiClient apiClient, BigDecimal commandId) {
+        LOGGER.debug("Waiting for Cloudera Manager to re-commission host with commandId: {}. [Server address: {}]", commandId, stack.getClusterManagerIp());
+        long timeout = POLL_FOR_5_MINUTES;
+        return pollCommandWithTimeListener(stack, apiClient, commandId, timeout,
+                new NoExceptionOnTimeoutClouderaManagerListenerTask(clouderaManagerApiPojoFactory, clusterEventService, "RecommissionHosts"));
```

Comment:
Do you mean? NoExceptionOnTimeoutClouderaManagerListenerTask
This was based on startPollingCmHostsDecommission

Reply:
Do you mean `SilentCMDecommissionHostListenerTask`? It is used only when all nodes targeted for decommission is already deleted on provider side and there is no change that we can decommission them and the result will be timeout anyway. (https://github.com/hortonworks/cloudbreak/commit/ed7dc64e3f657170228480ef410c27f40490d31a#diff-c8c6984e3f31ddc7f026766a133a5350d0ee25e65b7e2650623b7b6843c88c23) but in most of the use cases we should care about timeout from CM

Reply:
Went and looked at the usage of NoExceptionOnTimeoutClouderaManagerListenerTask again. It looks like the code which invokes the methods which use this (including the one using SilentCMDecommissionHostListenerTask) take care of handling Timeouts by checking the result of the task ClouderaManagerCommissioner for example checks `isTimeout(pollingResult)`... Am I wrong in this assumption? If so, what needs to change / which part is incorrect?

Reply:
Ok, i see you are handling timeout later with a general CloudbreakServiceException. Then I do not see why we need `NoExceptionOnTimeoutClouderaManagerListenerTask`? It is a general error prone class since it gives the opportunity for the developer to just ignore timeout. In this case we can simply use `ClouderaManagerDefaultListenerTask` and catch exception regarding timeout. But this should be not blocking in this case, so if it is possible, please create a ticket to remove this class if it is not necessary, since I see danger in using it.